src/Button.php

```diff
@@ -4,89 +4,263 @@
-        $new->encodeLabels = false;
+        $new->attributes['aria-expanded'] = $value === true ? 'true' : 'false';
 
         return $new;
     }
 
     /**
-     * The button label
+     * Sets the HTML attributes for the button component.
+     *
+     * @param array $values Attribute values indexed by attribute names.
+     *
+     * @return self A new instance with the specified attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      */
-    public function label(string $value): self
+    public function attributes(array $values): self
     {
+        $new = clone $this;
+        $new->attributes = array_merge($new->attributes, $values);
+
+        return $new;
+    }
+
+    /**
+     * Sets the 'data-bs-toggle' attribute for the button.
+     *
+     * @param string $value The value to set for the 'data-bs-toggle' attribute.
+     *
+     * @return self A new instance with the specified 'data-bs-toggle' value.
+     */
+    public function dataBsToggle(string $value): self
+    {
+        $new = clone $this;
+        $new->attributes['data-bs-toggle'] = $value;
+
+        return $new;
+    }
+
+    /**
+     * Sets the button to be disabled.
+     *
+     * @param bool $value Whether the button should be disabled.
+     *
+     * @return self A new instance with the button disabled.
+     */
+    public function disabled(bool $value = true): self
+    {
+        $new = clone $this;
+        $new->disabled = $value;
+
+        return $new;
+    }
+
+    /**
+     * Sets the ID of the button component.
+     *
+     * @param bool|string $value The ID of the button component. If `true`, an ID will be generated automatically.
+     *
+     * @return self A new instance with the specified ID.
+     */
+    public function id(bool|string $value): self
+    {
+        $new = clone $this;
+        $new->id = $value;
+
+        return $new;
+    }
+
+    /**
+     * The button label.
+     *
+     * @param string $value The label to display on the button.
+     * @param bool $encode Whether the label value should be HTML-encoded. Use this when rendering user-generated
+     * content to prevent XSS attacks.
+     *
+     * @return self A new instance with the specified label value.
+     */
+    public function label(string $value, bool $encode = true): self
+    {
+        if ($encode) {
+            $value = Html::encode($value);
+        }
+
         $new = clone $this;
         $new->label = $value;
 
         return $new;
     }
 
     /**
-     * The HTML attributes for the widget container tag. The following special options are recognized.
+     * Sets the button size to be large.
      *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return self A new instance with the button as a large button.
      */
-    public function options(array $value): self
+    public function large(): self
```

Comment:
Need ability to reset size. May be create method `size(?ButtonSize $value)` and enumeration `ButtonSize` ?

Reply:
You can change sizes via the `large()` or `small()` method, and if you don't want a size, just remove the method, don't create an enumeration because it's only two cases, and i think the syntax is more comfortable, but if you think the enumeration is better, we can create it.

Reply:
Agree, `large()` and `small()` methods are convient, but we syill need a method to base size. Let's add method `normalSize()` ?

Reply:
done.